### PR TITLE
[FAD-108] Create Pool Form and Image Uploader

### PR DESCRIPTION
**Pull request by sachasmart-weavik**
### Description

> Flow goes from artist dashboard to form, back to artist dashboard (still empty state - that's another ticket).

Closing [[FAD-95] ](https://github.com/weavik/free-artist-dao/pull/26)because of the significant changes to the image uploader. Better previews, styling, handling blobs, etc. 
### Checklist

- [ ] `CHANGELOG` has been updated (_if appropriate_)
- [ ] Environment variables updated (_if appropriate_)

[FAD-95]: https://weavi

https://user-images.githubusercontent.com/98773781/192410699-592ca58d-f838-4e82-8ea9-d70322c4dea1.mp4

k.atlassian.net/browse/FAD-95?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/client3/components/design-system/file-upload/file-upload.tsx [hunk 1]
```diff
@@ -14,48 +17,163 @@ interface FileUploadProps {
...
+    });
+
+    const addFile = async (fileToUpload: any) => {
+      if (!(fileToUpload instanceof Blob)) {
+        fileToUpload = new Blob([fileToUpload], { type: "application/json" });
+      }
+      const file = await addFileToIPFS(fileToUpload);
+      if (formContext !== null) {
+        if (file) {
+          setIpfsHash(file);
```
[the remark this answers is not in the document]
Comment (sachasmart-weavik) [reply packages/client3/components/design-system/file-upload/file-upload.tsx:56]: Yeap, good point. Thanks

File: packages/client3/components/design-system/file-upload/file-upload.tsx [hunk 2]
```diff
@@ -14,48 +17,163 @@ interface FileUploadProps {
...
+      onDrop: (acceptedFile) => {
+        setFile(
+          acceptedFile.map((file) =>
+            Object.assign(file, { preview: URL.createObjectURL(file) })
+          )
+        );
+      },
+    });
+
+    const addFile = async (fileToUpload: any) => {
```
[the remark this answers is not in the document]
Comment (sachasmart-weavik) [reply packages/client3/components/design-system/file-upload/file-upload.tsx:49]: Sure, I'll do that.

File: packages/client3/components/design-system/icon/icon.tsx [hunk 3]
```diff
@@ -100,6 +101,7 @@ export const Icon = forwardRef<SVGElement, IconProps>(function Icon(
       aria-hidden="true"
       ref={ref}
       className={clsx(sizeToClassName(size), "inline shrink-0", className)}
+      onClick={onClick}
```
Comment (luansievers) [inline packages/client3/components/design-system/icon/icon.tsx:104]: Hmm, I thought that every icon would be inside a button and we wouldn't need onClick on the icon, thoughts?

Comment (sachasmart-weavik) [reply packages/client3/components/design-system/icon/icon.tsx:104]: I don't really have any concerns to be honest. 

 @v1n33th, thoughts?

**Review by luansievers — APPROVED** [review]

**Pull request merged**